horizomer/utils/tree.py

```diff
@@ -44,6 +44,31 @@ def compare_topology(tree1, tree2):
     return n2p1 == n2p2
 
 
+def intersect_trees(tree1, tree2):
+    """Shrink two trees to contain only overlapping taxa.
+
+    Parameters
+    ----------
+    tree1 : skbio.TreeNode
+        first tree to intersect
+    tree2 : skbio.TreeNode
+        second tree to intersect
+
+    Returns
+    -------
+    tuple of two TreeNodes
+        resulting trees containing only overlapping taxa
+    """
+    taxa1 = set([tip.name for tip in tree1.tips()])
+    taxa2 = set([tip.name for tip in tree2.tips()])
+    taxa_lap = taxa1.intersection(taxa2)
+    if len(taxa_lap) == 0:
+        raise ValueError('Trees have no overlapping taxa.')
```

Comment:
why do you raise an error here? wouldn't it be more useful to return 0 and give a warning?

Reply:
That's because the return structure is a tuple of (TreeNode, TreeNode). User will typically do `x1, y1 = intersec_trees(x0, y0)`. To prevent further errors, better raise an error here.

Reply:
ok, fair.
ready to merge?

Reply:
No. Let me add something. @tkosciol